Cargo.toml

```diff
@@ -27,46 +27,37 @@ exclude = [
 ]
 
 [features]
-default = []
+default = ["compat-3-0-0", "rustls-tls", "dns-resolver"]
+compat-3-0-0 = []
 sync = []
-openssl-tls = ["openssl", "openssl-probe", "tokio-openssl"]
-
-# Enable support for v0.4 of the chrono crate in the public API of the BSON library.
-bson-chrono-0_4 = ["bson/chrono-0_4"]
-
-# Enable support for the serde_with crate in the BSON library.
-bson-serde_with = ["bson/serde_with"]
-
-# Enable support for v0.8 of the uuid crate in the public API of the BSON library.
-bson-uuid-0_8 = ["bson/uuid-0_8"]
-
-# Enable support for v1.x of the uuid crate in the public API of the BSON library.
-bson-uuid-1 = ["bson/uuid-1"]
+rustls-tls = ["dep:rustls", "dep:rustls-pemfile", "dep:tokio-rustls"]
+openssl-tls = ["dep:openssl", "dep:openssl-probe", "dep:tokio-openssl"]
+dns-resolver = ["dep:trust-dns-resolver", "dep:trust-dns-proto"]
 
 # Enable support for MONGODB-AWS authentication.
 # This can only be used with the tokio-runtime feature flag.
-aws-auth = ["reqwest"]
+aws-auth = ["dep:reqwest"]
```

Comment:
Using the `dep:foo` format prevents the optional dependencies from automatically becoming their own feature flags.

Reply:
nice, I think the need to do this will go away in the 2024 edition https://github.com/rust-lang/cargo/issues/12826